src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets

```diff
@@ -11,6 +11,8 @@
     <_CortypeFilePath Condition=" '$(_CortypeFilePath)' == '' ">$(MSBuildThisFileDirectory)..\inc\cortypeinfo.h</_CortypeFilePath>
     <_RexcepFilePath Condition=" '$(_RexcepFilePath)' == '' ">$(MSBuildThisFileDirectory)..\vm\rexcep.h</_RexcepFilePath>
     <_ILLinkDescriptorsIntermediatePath>$(IntermediateOutputPath)ILLink.Descriptors.Combined.xml</_ILLinkDescriptorsIntermediatePath>
+    <_DefineConstants>$(DefineConstants)</_DefineConstants>
+    <_DefineConstants Condition="'$(Configuration)' == 'Debug' or '$(Configuration)' == 'Checked'">$(DefineConstants);_DEBUG</_DefineConstants>
```

Comment:
How come this isn't handled by the existing infrastructure?

Reply:
Based on binlogs we only define `DEBUG` for managed code. `_DEBUG` seems to be only defined in native.

Reply:
Why do we need to define `_DEBUG` for anything managed?

Reply:
We should be able to just fix the corelib.h to use `DEBUG` instead of `_DEBUG`.